doc/managing_pachyderm/general_troubleshooting.md

```diff
@@ -0,0 +1,227 @@
+
+For example, to resolve this issue when Pachyderm is deployed to AWS, first find the node of which the pod is scheduled. In the output of the `kubectl describe po/pachd-xxx` command above, you should see the name of the node on which the pod is running.  In the AWS web console, find that node.. Once you have the right node, look in the bottom pane for the attached volume. Follow the link to the attached volume, and detach the volume. You may need to "Force Detach" it.
+
+Once it's detached (and marked as available). Restart the pod by killing it, e.g:
+
+```
+$kubectl delete po/pachd-xxx
+```
+
+It will take a moment for a new pod to get scheduled.
+
+---
+
+## Connecting to a Pachyderm Cluster
+
+- [Cannot connect via `pachctl` - context deadline exceeded](#cannot-connect-via-pachctl-context-deadline-exceeded)
+- [Certificate error when using `kubectl`](#certificate-error-when-using-kubectl)
+- [Uploads/downloads are slow](#uploads-downloads-are-slow)
+
+### Cannot connect via `pachctl` - context deadline exceeded
+
+#### Symptom
+
+You may be using the environmental variable `ADDRESS` to specify how `pachctl` talks to your Pachyderm cluster, or you may be forwarding the pachyderm port via `pachctl port-forward`.  In any event, you might see something similar to:
+
+```
+$ echo $ADDRESS
+1.2.3.4:30650
+$ pachctl version
+COMPONENT           VERSION                                          
+pachctl             1.4.8   
+context deadline exceeded
+```
+
+#### Recourse
+
+It's possible that the connection is just taking a while. Occasionally this can happen if your cluster is far away (deployed in a region across the country). Check your internet connection.
+
+It's also possible that you haven't poked a hole in the firewall to access the node on this port. Usually to do that you adjust a security rule (in AWS parlance a security group). For example, on AWS, if you find your node in the web console and click on it, you should see a link to the associated security group. Inspect that group. There should be a way to "add a rule" to the group. You'll want to enable TCP access (ingress) on port 30650. You'll usually be asked which incoming IPs should be whitelisted. You can choose to use your own, or enable it for everyone (0.0.0.0/0).
+
+
+### Certificate Error When Using Kubectl
+
+#### Symptom
+
+This can happen on any request using `kubectl` (e.g. `kubectl get all`), but it can also be seen when running `pachctl port-forward` because it uses `kubectl` under the hood. In particular you'll see:
+
+```
+$ kubectl version
+Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T20:41:24Z", GoVersion:"go1.8.1", Compiler:"gc", Platform:"darwin/amd64"}
+Unable to connect to the server: x509: certificate signed by unknown authority
+```
+
+#### Recourse
+
+Check if you're on any sort of VPN or other egress proxy that would break SSL.
+
```

Comment:
There's also the possibility that your credentials have expired. This just happened to me. In the case where you're using GKE and gcloud ... to renew the credentials do:

```
17-06-21[15:23:15]:stuck-job:130$kc get all
Unable to connect to the server: x509: certificate signed by unknown authority
17-06-21[15:23:29]:stuck-job:0$gcloud container clusters get-credentials my-cluster-name-dev
Fetching cluster endpoint and auth data.
kubeconfig entry generated for my-cluster-name-dev.
17-06-21[15:23:45]:stuck-job:0$kubectl config current-context
gke_my-org_us-east1-b_my-cluster-name-dev
```